arangod/IResearch/IResearchViewDBServer.cpp

```diff
@@ -382,35 +413,42 @@ arangodb::Result IResearchViewDBServer::rename(
   SCOPED_LOCK(mutex); // 'collections_' can be asynchronously modified
 
   for (auto& entry: _collections) {
-    auto res =  entry.second->rename(std::string(newName), doSync);
+    auto res =
+      entry.second->rename(generateName(newName, id(), entry.first), doSync);
 
     if (!res.ok()) {
       return res; // fail on first failure
     }
   }
 
+  static const std::function<bool(irs::string_ref const& key)> acceptor = [](
+      irs::string_ref const& key
+  )->bool {
+    return key != NAME_FIELD; // ignore name filed
+  };
   arangodb::velocypack::Builder builder;
 
   builder.openObject();
+  builder.add(NAME_FIELD, arangodb::velocypack::Value(std::move(newName)));
 
-  if (!mergeSlice(builder, _meta.slice())) {
+  if (!mergeSliceSkipKeys(builder, _meta.slice(), acceptor)) {
```

Comment:
Don't we have logic like this in VPackCollection or in the VelocyPackHelpers

Reply:
IIRC the version in VelocyPackHelpers internally creates a new arangodb::velocypack::Builder instead of simply not adding keys to the existing builder